From 5f744958bd2a28fa8e7e31102fdee6826b692f67 Mon Sep 17 00:00:00 2001
From: Rui Loura <rui@joyent.com>
Date: Wed, 6 Feb 2019 19:44:21 +0000
Subject: [PATCH] TRITON-875 fw-agent adminIP functions need to be factored out
 Reviewed by: Jason King <jason.king@joyent.com> Reviewed by: Cody Peter Mello
 <melloc@writev.io> Approved by: Cody Peter Mello <melloc@writev.io>

---
 lib/config.js | 22 ----------------------
 main.js       |  3 ++-
 package.json  |  3 ++-
 3 files changed, 4 insertions(+), 24 deletions(-)

diff --git a/lib/config.js b/lib/config.js
index 7709446..daba76e 100644
--- a/lib/config.js
+++ b/lib/config.js
@@ -22,27 +22,6 @@ var VError = require('verror').VError;
 // --- Exports
 
 
-function findSysinfoAdminIP(si) {
-    var iface;
-    var ifaces = si['Network Interfaces'];
-    var tag = 'admin';
-
-    if (si['Admin NIC Tag']) {
-        tag = si['Admin NIC Tag'];
-    }
-
-    for (var i in ifaces) {
-        iface = ifaces[i];
-
-        if (iface['NIC Names'] && iface['NIC Names'].indexOf(tag) !== -1) {
-            return iface.ip4addr;
-        }
-    }
-
-    return null;
-}
-
-
 function getSysinfoValue(name, callback) {
     sysinfo(function (err, conf) {
         if (err) {
@@ -95,7 +74,6 @@ function sysinfo(callback) {
 
 
 module.exports = {
-    findSysinfoAdminIP: findSysinfoAdminIP,
     getSysinfoValue: getSysinfoValue,
     sdc: config,
     sysinfo: sysinfo
diff --git a/main.js b/main.js
index 124badd..455b5e2 100644
--- a/main.js
+++ b/main.js
@@ -19,6 +19,7 @@ var bunyan = require('bunyan');
 var config = require('./lib/config');
 var firewaller = require('./lib/agent');
 var fs = require('fs');
+var netconfig = require('triton-netconfig');
 var path = require('path');
 var VError = require('verror').VError;
 
@@ -72,7 +73,7 @@ config.sdc(function (err, sdcConfig) {
         CONFIG.imageVersion = image;
         CONFIG.fwapi.host = sdcConfig.fwapi_domain;
         CONFIG.vmapi = { host: sdcConfig.vmapi_domain };
-        CONFIG.listenIP = config.findSysinfoAdminIP(sysinfo);
+        CONFIG.listenIP = netconfig.adminIpFromSysinfo(sysinfo);
 
         if (!CONFIG.listenIP) {
             LOG.error({ sysinfo: sysinfo }, 'Error finding sysinfo admin IP');
diff --git a/package.json b/package.json
index 89d9a9b..dc7a0d8 100644
--- a/package.json
+++ b/package.json
@@ -1,7 +1,7 @@
 {
     "name": "firewaller",
     "description": "Triton Data Center firewalling agent",
-    "version": "1.5.2",
+    "version": "1.5.3",
     "author": "Joyent (joyent.com)",
     "private": true,
     "dependencies": {
@@ -14,6 +14,7 @@
         "lockfd": "~1.2.1",
         "restify": "2.6.1",
         "sdc-clients": "12.1.1",
+        "triton-netconfig": "1.3.0",
         "uuid": "3.0.1",
         "vasync": "1.6.3",
         "verror": "1.3.6"
-- 
2.21.0

